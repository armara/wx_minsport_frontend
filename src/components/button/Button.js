import React, { useCallback } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import 'components/button/Button.scss'

const Button = ({
  id,
  children,
  type,
  className,
  onClick,
  icon,
  size,
  disabled,
  title,
  element: Element,
}) => {
  const memoizedOnClick = useCallback(
    event => {
      if (!disabled) {
        onClick(event)
      }
    },
    [onClick, disabled]
  )
  return (
    <Element
      id={id}
      type={type}
      className={classnames({
        [size]: size,
        [className]: className,
        button: true,
        'base-button': true,
        disabled,
      })}
      onClick={memoizedOnClick}
      disabled={disabled}
      title={title}
    >
      {icon && (
        <span className={classnames(['icon', size])}>
          <i className={icon} />
        </span>
      )}
      {children}
    </Element>
  )
}

Button.defaultProps = {
  id: null,
  children: '',
  className: 'button',
  icon: '',
  size: '',
  type: 'button',
  onClick: () => {},
  element: 'button',
  disabled: false,
  title: '',
}

Button.propTypes = {
  id: PropTypes.string,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  icon: PropTypes.string,
  size: PropTypes.oneOf(['is-small', '', 'is-medium', 'is-large']),
  type: PropTypes.oneOf(['button', 'submit', 'reset']),
  onClick: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
  ]),
  element: PropTypes.oneOf(['button', 'a']),
  title: PropTypes.string,
}

export default Button
