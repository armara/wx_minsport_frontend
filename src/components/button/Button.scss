@import "styles/mixins";

.base-button {
  cursor: pointer;

  &.button {
    .icon:first-child {
      margin-right: 7px;
    }
  }

  &.disabled {
    opacity: 0.3;
    outline: none;
    border: none;
  }

  &.nav-green-button {
    @include nav-header-text('12px','18px',$white);
    background-color: $nestea-light;
    height: 30px;
    padding-left: 20px;
    padding-right: 20px;

    &:hover,
    &:focus {
      background-color: $nestea;
      color: $white;
    }
  }

  &.grey-button {
    @include text(9px, normal, $white, normal);
    background-color: $soft-steel;
    border-radius: 4px;
    height: 30px;
    width: 135px;
  }

  &.nav-grey-button {
    @include nav-header-text('9px','18px',$white);
    background-color: $soft-steel;
    height: 30px;
    padding-left: 20px;
    padding-right: 20px;

    &:hover,
    &:focus {
      color: $white;
    }
  }

  &.inline-button {
    @include add-inline-button();
  }

  &.grey-control-button {
    background-color: $soft-steel;
    margin-left: 10px;
  }

  &.control-button-save {
    background-color: $nestea-light;

    &:hover {
      background-color: $nestea;
    }
  }

  &.control-buttons_cancel {
    background-color: $fog-gray;

    &:hover {
      background-color: $brown-gray-light;
    }
  }

  &.control-buttons_cancel,
  &.grey-control-button,
  &.control-button-save {
    @include control-button();
  }

  &.grey-control-button:hover,
  &.grey-control-button:focus,
  &.control-button-save:hover,
  &.control-button-save:focus,
  &.control-buttons_cancel:hover,
  &.control-buttons_cancel:focus {
    color: $white;
  }

  &.transparent-button {
    @include nav-header-text('12px','18px',$black);
    background-color: $white;
    border: none;
  }
  
	&.btn-add-training,
	&.btn-add-event {
	  border: none;
	  @include page-header-text('13px','unset');    
	}

	&.btn-add-training,
	&.btn-add-training:hover {
	  color: $android-green;    
	}
	&.btn-add-event,
	&.btn-add-event:hover {
	  color: $light-amber;
	}
}