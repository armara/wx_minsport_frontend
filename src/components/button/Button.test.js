import React from 'react'
import { configure, mount, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Button from 'components/button/Button'

configure({ adapter: new Adapter() })

describe('<Button />', () => {
  it('should be rendered without throwing an error', () => {
    expect(
      mount(<Button>Push the button</Button>).contains('Push the button')
    ).toBe(true)
  })

  it('renders correctly default props', () => {
    const tree = renderer
      .create(
        <Button>
          <p>Peter Great Embassy</p>
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should be selectable by class ".button"', () => {
    expect(mount(<Button>Push the button</Button>).is('.button')).toBe(true)
  })

  it('should be selectable by "button" tag by default', () => {
    expect(mount(<Button />).find('button').length).toBe(1)
  })

  it('should be selectable by "a" tag if specified', () => {
    expect(mount(<Button element="a" />).find('a').length).toBe(1)
  })

  it('should have "button" type by default', () => {
    expect(mount(<Button />).prop('type')).toBe('button')
  })

  it('should have "submit" type if specified', () => {
    expect(mount(<Button type="submit" />).prop('type')).toBe('submit')
  })

  it('should mount in a full DOM', () => {
    expect(
      mount(<Button>Push the button</Button>).find('.base-button').length
    ).toBe(1)
  })

  it('should render to static HTML', () => {
    expect(render(<Button>Push the button</Button>).text()).toEqual(
      'Push the button'
    )
  })

  it('has selectable icon', () => {
    const button = mount(<Button icon="ic-search" />)
    expect(button.find('span').length).toBe(1)
    expect(button.find('span').find('.ic-search').length).toBe(1)
  })

  it('does not have icon by default', () => {
    expect(mount(<Button />).find('.icon').length).toBe(0)
  })

  it('triggers onClick correctly', () => {
    const mockCallback = jest.fn()
    const button = mount(
      <Button onClick={mockCallback}>Push the button</Button>
    )
    button.find('.base-button').simulate('click')
    expect(mockCallback.mock.calls.length).toEqual(1)
  })
})
