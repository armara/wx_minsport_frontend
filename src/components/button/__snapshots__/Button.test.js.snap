// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly default props 1`] = `
<button
  className="button base-button"
  disabled={false}
  id={null}
  onClick={[Function]}
  title=""
  type="button"
>
  
  <p>
    Peter Great Embassy
  </p>
</button>
`;
